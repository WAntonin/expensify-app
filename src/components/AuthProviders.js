import React from 'react'
import { connect } from 'react-redux'
import { startGoogleLogin, startFacebookLogin } from '../actions/auth'

export class AuthProviders extends React.Component {
    render() {
        return (
            <div >
                <button className="button button--login" onClick={this.props.startGoogleLogin}>
                    <img className="button--login__item" src="/images/google_icon.png" />
                    <span>Sign in with Google</span>
                </button>
                <button className="button button--login" onClick={this.props.startFacebookLogin}>
                    <img className="button--login__item" src="/images/facebook_icon.png" />
                    <span>Sign in with Facebook</span>
                </button>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startFacebookLogin: () => dispatch(startFacebookLogin())
})

export default connect(undefined, mapDispatchToProps)(AuthProviders)