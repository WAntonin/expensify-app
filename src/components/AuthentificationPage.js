import React from 'react'
import { connect } from 'react-redux'

export class AuthentificationPage extends React.Component {
    render() {
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">Expensify App</h1>
                    <h2 className="box-layout__subtitle">An application to log your expenses</h2>
                </div>
            </div>
        )
    }
}