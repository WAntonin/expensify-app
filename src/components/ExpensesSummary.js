import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import selectTotalExpenses from '../selectors/expenses-total'
import numeral from 'numeral'


export const ExpensesSummary = ({expenseCount, expenseTotal}) => {
    const plural = expenseCount > 1 ? 's' : ''
    const formattedExpenseTotal = numeral(expenseTotal / 100).format('0,0.00$')
    return(
    <div>
        <h1>Viewing {expenseCount} expense{plural} for a total of {formattedExpenseTotal}</h1>
    </div>
    )
}

const mapStateToProps = (state) => {
    const filteredExpenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: filteredExpenses.length,
        expenseTotal: selectTotalExpenses(filteredExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)