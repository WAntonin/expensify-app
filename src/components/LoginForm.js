import React from 'react'
import isEmail from 'validator/lib/isEmail'
import { startEmailPasswordLogin, startGoogleLogin, startFacebookLogin } from '../actions/auth'
import { connect } from 'react-redux';

export class LoginForm extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            email: '',
            password: '',
            error: ''
        }
    }
    onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }
    onSubmit = (e) => {
        e.preventDefault()
        if (!isEmail(this.state.email) || this.state.password.length < 6) {
            this.setState(() => ({ error: 'Invalid password or email' }))
        } else {
            this.setState(() => ({ error: '' }))
            this.props.startEmailPasswordLogin(this.state.email, this.state.password)
        }
    }
    render() {
        const isInvalid = this.state.password === '' || this.state.email === ''
        return (
            <div className="box-layout__box">
                <button className="button button--login" onClick={this.props.startGoogleLogin}>
                    <img className="button--login__item" src="/images/google_icon.png" />
                    <span>Sign in with Google</span>
                </button>
                <button className="button button--login" onClick={this.props.startFacebookLogin}>
                    <img className="button--login__item" src="/images/facebook_icon.png" />
                    <span>Sign in with Facebook</span>
                </button>
                <h3>OR</h3>
                <form onSubmit={this.onSubmit}>
                    <input
                        type="email"
                        value={this.state.email}
                        name="email"
                        onChange={this.onChange}
                        placeholder="email"
                        autoComplete="on"
                    />
                    <input
                        type="password"
                        value={this.state.password}
                        name="password"
                        onChange={this.onChange}
                        placeholder="password (6 char. min.)"

                        autoComplete="on"
                    />
                    <button disabled={isInvalid} type="submit">Sign In</button>
                </form>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startFacebookLogin: () => dispatch(startFacebookLogin()),
    startEmailPasswordLogin: (email, password) => dispatch(startEmailPasswordLogin(email, password))
})

export default connect(undefined, mapDispatchToProps)(LoginForm)