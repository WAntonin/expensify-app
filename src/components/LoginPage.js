import React from 'react'
import { connect } from 'react-redux'
import { startGoogleLogin, startFacebookLogin } from '../actions/auth'

export const LoginPage = ({ startGoogleLogin, startFacebookLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify App</h1>
            <p>An application to log your expenses</p>
            <button className="button button--login" onClick={startGoogleLogin}>
                <img className="button--login__item" src="/images/google_icon.png" />
                <span>Login with Google</span>
            </button>
            <button className="button button--login" onClick={startFacebookLogin}>
                <img className="button--login__item" src="/images/facebook_icon.png" />
                <span>Login with Facebook</span>
            </button>
        </div>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
    startGoogleLogin: () => dispatch(startGoogleLogin()),
    startFacebookLogin: () => dispatch(startFacebookLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)