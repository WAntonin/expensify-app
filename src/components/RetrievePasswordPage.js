import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import isEmail from 'validator/lib/isEmail'
import { startPasswordReset } from '../actions/auth'

export class RetrievePasswordPage extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            email: props.userId ? props.userId.email : '',
            error: props.userId ? props.userId.error : ''
        }
    }
    onEmailChange = (e) => {
        console.log(e.target.value)
        const email = e.target.value
        this.setState(() => ({ email }))
    }
    onSubmit = (e) => {
        e.preventDefault()
        if (isEmail(this.state.email)) {
            this.props.startPasswordReset(this.state.email).catch((error) => {
                this.setState({ error: error.code })
            })
            console.log(this.state.error)
        }
    }
    render() {
        const isInvalid = !isEmail(this.state.email)
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <form className="form" onSubmit={this.onSubmit}>
                        <input
                            className="text-input"
                            type="email"
                            value={this.state.email}
                            name="email"
                            onChange={this.onEmailChange}
                            placeholder="email"
                            autoComplete="on"
                        />
                        {
                            this.state.error === 'auth/user-not-found'
                            && <p className="form__error">Invalid Email</p>
                        }
                        <button className="button" disabled={isInvalid} type="submit">Reset my password</button>
                        <Link to="/">Back to Sign In page</Link>
                    </form>
                </div>
            </div>
        )
    }
}
// const mapStateToProps = (state) => ({
//     email: state.email
// })
const mapDispatchToProps = (dispatch) => ({
    startPasswordReset: (email) => dispatch(startPasswordReset(email))
})

export default connect(undefined, mapDispatchToProps)(RetrievePasswordPage)