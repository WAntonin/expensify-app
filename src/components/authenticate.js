import React from 'react'

const authenticate = (WrappedComponent) => {
    class Authenticate extends React.Component {
        constructor(props) {
            super(props)
            this.state = {
                email: '',
                password: ''
            }
            this.onInputChange = this.onInputChange.bind(this);
            this.submit = this.submit.bind(this)
        }
        onInputChange = (e) => {
            e.preventDefault()
            console.log('INPUT_CHANGE')
            const inputName = e.target.name 
            const inputValue = e.target.value 
            this.setState({ [inputName]: inputValue})
        }
        submit = () => {
            console.log('SUBMIT')
            this.props.onSubmit({
                email: this.state.email,
                password: this.state.password
            })
            console.log('SUBMIT', this.state.email, this.state.password)
            debugger
        }
        render() {
            const { onChange, onSubmit, ...otherProps } = this.props
            return (
                <WrappedComponent
                onChange={ this.onInputChange}
                onSubmit={this.submit}
                {...otherProps}
                />)
        }
    }
    return Authenticate
}

export default authenticate