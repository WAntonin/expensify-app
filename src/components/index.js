import React from 'react'
import { connect } from 'react-redux'
import AuthProviders from './AuthProviders'
import SignIn from './SignIn'
import SignUp from './SignUp'
import { startEmailPasswordLogin, startCreateUserAccount } from '../actions/auth'
import authenticate from './authenticate'

const SignInForm = authenticate(SignIn)
const SignUpForm = authenticate(SignUp)


export class Authenticate extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            isSignIn: true
        }
        this.toggleForm = this.toggleForm.bind(this)
    }
    toggleForm = () => {
        this.setState(() => ({
            isSignIn: !this.state.isSignIn
        }))
    }
    render() {
        const { startCreateUserAccount, startEmailPasswordLogin } = this.props
        return (
            // <div className="box-layout">
            //     <div className="box-layout__box">
            //         <h1 className="box-layout__title">Expensify App</h1>
            //         <h2 className="box-layout__subtitle">An application to log your expenses</h2>
            //     </div>
            // </div>
            <div className="box-layout">
                <div className="box-layout__box">
                    <AuthProviders />
                </div>
                {
                    this.state.isSignIn ? (
                        <SignInForm
                            onSubmit={startEmailPasswordLogin}
                            showSignUp={this.toggleForm}
                        />
                    ) : 
                    (
                        <SignUpForm
                            onSubmit={startCreateUserAccount}
                            showSignIn={this.toggleForm}
                        />
                    )
                }
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    startCreateUserAccount: (email, password) => dispatch(startCreateUserAccount(email, password)),
    startEmailPasswordLogin: (email, password) => dispatch(startEmailPasswordLogin(email, password))
})

export default connect(undefined, mapDispatchToProps)(Authenticate)