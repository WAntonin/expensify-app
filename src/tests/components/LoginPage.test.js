import React from 'react'
import { shallow } from 'enzyme'
import { LoginPage } from '../../components/LoginPage'
import userId from '../fixtures/userId.js'


test('should render LoginPage correctly', () => {
    const wrapper = shallow(<LoginPage />)
    expect(wrapper).toMatchSnapshot()
})

test('should handle startEmailPasswordLogin', () => {
    const startEmailPasswordLogin = jest.fn()
    const wrapper = shallow(<LoginPage 
        startEmailPasswordLogin={startEmailPasswordLogin}
        userId={userId}
        />)
    wrapper.find('LoginForm').prop('onSubmit')(userId)
    expect(startEmailPasswordLogin).toHaveBeenLastCalledWith(userId.email, userId.password)
})