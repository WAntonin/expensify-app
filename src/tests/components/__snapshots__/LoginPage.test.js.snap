// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify App
    </h1>
    <h2
      className="box-layout__subtitle"
    >
      An application to log your expenses
    </h2>
  </div>
  <div
    className="box-layout__box"
  >
    <Connect(AuthProviders) />
    <h3
      className="box-layout__subtitle box-layout__subtitle--separator"
    >
      <span>
        OR
      </span>
    </h3>
    <LoginForm
      onSubmit={[Function]}
    />
  </div>
  <Link
    className="box-layout__box box-layout__box--link"
    replace={false}
    to="/signup"
  >
    <p
      className="box-layout__message"
    >
      If you don't have an account you can sign up here.
    </p>
  </Link>
</div>
`;
